exonum/src/storage/keys.rs

```diff
@@ -115,110 +111,64 @@ impl StorageKey for u8 {
     }
 }
 
-/// Uses big-endian encoding.
-impl StorageKey for u16 {
-    fn size(&self) -> usize {
-        2
-    }
-
-    fn write(&self, buffer: &mut [u8]) {
-        BigEndian::write_u16(buffer, *self)
-    }
-
-    fn read(buffer: &[u8]) -> Self {
-        BigEndian::read_u16(buffer)
-    }
-}
-
-/// Uses big-endian encoding.
-impl StorageKey for u32 {
-    fn size(&self) -> usize {
-        4
-    }
-
-    fn write(&self, buffer: &mut [u8]) {
-        BigEndian::write_u32(buffer, *self)
-    }
-
-    fn read(buffer: &[u8]) -> Self {
-        BigEndian::read_u32(buffer)
-    }
-}
-
-/// Uses big-endian encoding.
-impl StorageKey for u64 {
-    fn size(&self) -> usize {
-        8
-    }
-
-    fn write(&self, buffer: &mut [u8]) {
-        BigEndian::write_u64(buffer, *self)
-    }
-
-    fn read(buffer: &[u8]) -> Self {
-        BigEndian::read_u64(buffer)
-    }
-}
-
-/// **Not sorted in the natural order.**
+/// Uses encoding with the values mapped to `u8`
+/// by adding the corresponding constant (`128`) to the value.
 impl StorageKey for i8 {
     fn size(&self) -> usize {
         1
     }
 
     fn write(&self, buffer: &mut [u8]) {
-        buffer[0] = *self as u8
+        buffer[0] = self.wrapping_add(i8::min_value()) as u8;
     }
 
     fn read(buffer: &[u8]) -> Self {
-        buffer[0] as i8
+        buffer[0].wrapping_add(i8::min_value() as u8) as i8
```

Comment:
Nit: I would expect `write` and `read` to be complimentary and to use `add` / `sub`, even though the effect should be exactly the same.